feat(llc): add support for sharing live and static locations

xsahil03x opened the pull request:
# Submit a pull request
<!--Internal tickets have to be added by Stream devs-->
Linear: FLU-197

## Screenshots / Videos

<!-- Consider to add screenshots and/or videos to show the effect of the change -->


https://github.com/user-attachments/assets/9bd71bf7-2da5-4a8f-a689-a680e4962cbc



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Introduced location sharing in chat, allowing users to send static or live locations in messages.
  * Added interactive map views and live location updates within chat messages and threads.
  * Users can stop sharing live locations directly from the chat interface.
  * New dialogs for picking and viewing shared locations, with support for sharing duration selection.
  * Enhanced message rendering to display and interact with location attachments.

* **Bug Fixes**
  * Corrected initialization order in channel list page state.

* **Chores**
  * Added required location permissions and dependencies for Android, iOS, and macOS platforms.
  * Updated configuration files to support location features and new sample user.

* **Documentation**
  * Updated public exports and configuration to reflect new location models and capabilities.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

xsahil03x committed:
feat(llc): add support for sharing live and static locations

xsahil03x committed:
chore: minor improvements

xsahil03x committed:
chore: run formatter

xsahil03x committed:
chore: minor improvement

xsahil03x committed:
chore: improve location attachment option

xsahil03x committed:
chore: apply review suggestion

xsahil03x committed:
chore: remove dead code

xsahil03x committed:
chore: add client.dart tests

xsahil03x committed:
chore: add channel.dart tests

xsahil03x committed:
test: fix failing tests

xsahil03x committed:
chore: add user_api_test.dart

xsahil03x committed:
chore: update CHANGELOG.md

xsahil03x committed:
chore: add event_resolvers and event_controller test

xsahil03x committed:
chore: fix analysis

xsahil03x committed:
chore: add location_test.dart

xsahil03x committed:
chore: cleanup and minor fixes

xsahil03x committed:
chore: fix test

renefloor commented on packages/stream_chat/lib/src/client/client.dart:
```diff
@@ -2174,6 +2234,103 @@ class ClientState {
     );
   }
 
+  void _listenLocationShared() {
```
The client and channel files are very large. I wonder if we for example can put all location related code to separates files.

xsahil03x replied:
Yes, we should move all the event handling logic outside of state into another file but let's do it in a separate ticket.

renefloor commented on sample_app/lib/pages/channel_page.dart:
```diff
@@ -124,12 +130,77 @@ class _ChannelPageState extends State<ChannelPage> {
+  ) async {
+    if (result.endSharingAt case final endSharingAt?) {
+      return channel.startLiveLocationSharing(
+        createdByDeviceId: 'test-device-id',
```
Do we want to create a random UUID and store this locally in shared prefs or something, so we can identify devices?

xsahil03x replied:
The `createdByDeviceId` is going to be made optional in backend soon, There's a PR yet to be merged. I will remove this once the backend is deployed.

xsahil03x replied:
Removed

xsahil03x committed:
Merge remote-tracking branch 'origin/v10.0.0' into feat/location-attachment

# Conflicts:
#	packages/stream_chat/lib/src/core/api/responses.dart
#	packages/stream_chat/lib/src/core/api/responses.g.dart
#	packages/stream_chat/lib/src/core/api/user_api.dart
#	packages/stream_chat/test/src/core/api/user_api_test.dart

xsahil03x committed:
refactor(llc): group event listeners using `region` comments

Groups related event listener method calls within `StreamChatClient` and `Channel` using `// region` and `// endregion` comments for better code organization and readability.

xsahil03x committed:
chore: format

xsahil03x committed:
refactor: make `createdByDeviceId` optional in location-related methods

xsahil03x commented:
@coderabbitai review

xsahil03x committed:
feat: add `locationLabel` for location attachment

Adds a new translatable string `locationLabel` to represent location attachments in the UI.

This change includes:
- Adding `locationLabel` to `Translations` and `DefaultTranslations` in `stream_chat_flutter`.
- Updating `StreamMessagePreviewText` and `QuotedMessageWidget` to use the new `locationLabel`.
- Adding translations for `locationLabel` to all supported languages in `stream_chat_localizations`.
- Updating tests and example in `stream_chat_localizations`.
